### PR TITLE
Update rlvm ports

### PR DESCRIPTION
**Pull request by JeodC**
- Mark old rlvm port experimental
- Add rlvm runtime
- Add games as individual ports

This port was already released but is receiving a much needed update. The sdl12-compat library and gl4es use newer versions, the launchscript meets current standards, gptokeyb is more user-friendly, and some [problems with rlvm were fixed](https://github.com/JeodC/rlvm/commits/master/).

Tested on RG552 with amberelec with Clannad HD. Port is now good for widescreen devices, should also be fine on 4:3 devices.

All replacement binaries and libraries were compiled under bullseye for arkos compatibility.

### CODE REVIEW

File: ports/clannad/Clannad.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+# Setup controls
+$ESUDO chmod 666 /dev/tty0
+$ESUDO chmod 666 /dev/tty1
+$ESUDO chmod 666 /dev/uinput
+$GPTOKEYB "$runtime" -c "rlvm.gptk" & 
+
+# Run the game
+echo "Loading, please wait... (might take a while!)" > /dev/tty0
+$runtime $font "$GAMEDIR/gamedata" 2>&1 | tee ./"log.txt"
```
Comment (Cebion) [inline ports/clannad/Clannad.sh:68]: Old logging method,
```suggestion
$runtime $font "$GAMEDIR/gamedata"
```

File: ports/clannad/Clannad.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+# Set variables
+GAMEDIR="/$directory/ports/clannad"
+DEVICE_ARCH="${DEVICE_ARCH:-aarch64}"
+runtime="rlvm"
+rlvm_dir="$HOME/rlvm"
+rlvm_file="$controlfolder/libs/${runtime}.squashfs"
+font="--font $rlvm_dir/fonts/msgothic.ttc"
+
+cd $GAMEDIR
+
```
Comment (Cebion) [inline ports/clannad/Clannad.sh:29]:
```suggestion
> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
```

File: runtimes/rlvm.squashfs.md5
Comment (kloptops) [inline runtimes/rlvm.squashfs.md5:1]: File isnt needed.

File: runtimes/rlvm.squashfs
Comment (kloptops) [inline runtimes/rlvm.squashfs:1]: rename `rlvm.aarch64.squashfs`

add entry to runtimes.json:
```json
    "rlvm.squashfs":  {
        "name": "Real Live VM",
        "default": "aarch64",
        "arch": {
            "aarch64": "rlvm.aarch64.squashfs"
        }
    },
```

**Review by Cebion — CHANGES_REQUESTED** [review]

File: ports/clannad/port.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    ],
+    "desc": "The story follows the life of Tomoya Okazaki, a high school delinquent who meets many people in his last year at school, including five girls, and helps resolve their individual problems. Follows a branching plot line which offers pre-determined scenarios with courses of interaction, and focuses on the appeal of the five female main characters by the player character.",
+    "inst": "Add your game assets to `ports/clannad/gamedata`.",
+    "genres": [
+      "visual-novel"
+    ],
+    "image": "screenshot.png",
+    "rtr": false,
+    "exp": false,
+    "runtime": "rlvm",
```
Comment (kloptops) [inline ports/clannad/port.json:23]: This should be `rlvm.squashfs`

**Review by kloptops — CHANGES_REQUESTED** [review]
Not sure if RLVM should be removed as it still can be used to run some of the other games. But maybe rename it "Rlvm (deprecated)" as the title in the port.json.

File: runtimes/rlvm.squashfs
Comment (JeodC) [reply runtimes/rlvm.squashfs:1]: Does that matter now since we don't have any armhf-only devices anymore?

File: ports/clannad/port.json (hunk 3 above)
Comment (JeodC) [reply ports/clannad/port.json:23]: > This should be `rlvm.squashfs`

Not `rlvm.aarch64.squashfs`?

File: ports/clannad/Clannad.sh [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+source $controlfolder/device_info.txt
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+get_controls
+
+# Set variables
+GAMEDIR="/$directory/ports/clannad"
+DEVICE_ARCH="${DEVICE_ARCH:-aarch64}"
+runtime="rlvm"
+rlvm_dir="$HOME/rlvm"
+rlvm_file="$controlfolder/libs/${runtime}.${DEVICE_ARCH}.squashfs"
```
Comment (kloptops) [inline ports/clannad/Clannad.sh:25]: The runtime file will be `rlvm.squashfs`, not `rlvm.aarch64.squashfs`.
```suggestion
rlvm_file="$controlfolder/libs/${runtime}.squashfs"
```

File: ports/clannad/Clannad.sh [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+# Check for runtime
+if [ ! -f "$rlvm_file" ]; then
+  # Check for runtime if not downloaded via PM
+  if [ ! -f "$controlfolder/harbourmaster" ]; then
+    echo "This port requires the latest PortMaster to run, please go to https://portmaster.games/ for more info." > /dev/tty0
+    sleep 5
+    exit 1
+  fi
+  $ESUDO $controlfolder/harbourmaster --quiet --no-check runtime_check "${runtime}.${DEVICE_ARCH}.squashfs"
```
Comment (kloptops) [inline ports/clannad/Clannad.sh:46]:
```suggestion
  $ESUDO $controlfolder/harbourmaster --quiet --no-check runtime_check "${runtime}.squashfs"
```

**Review by kloptops — CHANGES_REQUESTED** [review]
Just need to fix the one bash script.

**Review by kloptops — CHANGES_REQUESTED** [review]

File: ports/rlvm/rlvm/gl4es.armhf/libEGL.so.1
[the remark this answers is not in the document]
Comment (JeodC) [reply ports/rlvm/rlvm/gl4es.armhf/libEGL.so.1:1]: They're not?

**Comment by JeodC** [comment]
Ready for re-review. Air and Kanon require 4:3. Others working fine on widescreen.

**Review by kloptops — APPROVED** [review]

File: ports/rlvm/port.json [hunk 6]
```diff
@@ -21,6 +21,7 @@
         "rtr": false,
 		"exp": true,
         "runtime": null,
+        "exp": true,
```
Comment (kloptops) [inline ports/rlvm/port.json:24]: Whoops, `"exp": true,` tag is in there twice. 😁 
```suggestion
```

**Review by kloptops — CHANGES_REQUESTED** [review]

**Comment by JeodC** [comment]
Ok to merge

**Review by kloptops — DISMISSED** [review]

File: ports/air/README.md
Comment (Cebion) [inline ports/air/README.md:1]: Compile instructions missing.

File: ports/air/port.json [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      "visual novel"
+    ],
+    "image": "screenshot.png",
+    "rtr": false,
+    "exp": false,
+    "runtime": "rlvm.squashfs",
+	"arch": [
+		"aarch64"
+	],
+    "reqs": [["4:3"]]
```
Comment (Cebion) [inline ports/air/port.json:27]: Does this really only work on 4:3 devices?

**Review by Cebion — CHANGES_REQUESTED** [review]

**Comment by Cebion** [comment]
Also needs testing matrix since there is a new runtime and libraries. 
So tests on all major CFWs and devices as usual =)

File: ports/air/port.json (hunk 7 above)
Comment (JeodC) [reply ports/air/port.json:27]: Widescreen devices cannot bring up the menu to save and load games for air or kanon.

File: ports/air/README.md
Comment (JeodC) [reply ports/air/README.md:1]: https://github.com/JeodC/rlvm

Solarus does not have compile instructions nor do any other runtimes to my knowledge. If you're going to enforce a new standard make sure the existing ports follow it first.

**Review by kloptops — CHANGES_REQUESTED** [review]
A few more changes. 😁

File: ports/air/Air.sh [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+# Create the config folders
+for SAVEDIR in "${SAVEDIR[@]}"; do
+    rm -rf "$HOME/.rlvm/$SAVEDIR"
+    ln -s "$GAMEDIR/saves" "$HOME/.rlvm/$SAVEDIR"
+done
+
+export LD_LIBRARY_PATH="$rlvm_dir/libs":$LD_LIBRARY_PATH
+if [ "$LIBGL_FB" != "" ]; then
+  export SDL_VIDEO_GL_DRIVER="$rlvm_dir/gl4es/libGL.so.1"
+fi
```
Comment (Cebion) [inline ports/air/Air.sh:64]:
```suggestion
  export LD_LIBRARY_PATH="$rlvm_dir/gl4es:$LD_LIBRARY_PATH"
fi
```

File: ports/clannad/Clannad.sh [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+$ESUDO mount "$rlvm_file" "$rlvm_dir"
+PATH="$rlvm_dir:$PATH"
+
+# Create config dir
+rm -rf "$HOME/.rlvm/KEY_CLANNAD_ENHD"
+ln -s "$GAMEDIR/saves" "$HOME/.rlvm/KEY_CLANNAD_ENHD"
+
+export LD_LIBRARY_PATH="$rlvm_dir/libs":$LD_LIBRARY_PATH
+if [ "$LIBGL_FB" != "" ]; then
+  export SDL_VIDEO_GL_DRIVER="$rlvm_dir/gl4es/libGL.so.1"
```
Comment (Cebion) [inline ports/clannad/Clannad.sh:61]:
```suggestion
  export SDL_VIDEO_GL_DRIVER="$rlvm_dir/gl4es/libGL.so.1"
  export LD_LIBRARY_PATH="$rlvm_dir/gl4es:$LD_LIBRARY_PATH"
```

File: ports/kanon/Kanon.sh [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+# Create the config folders
+for SAVEDIR in "${SAVEDIR[@]}"; do
+    rm -rf "$HOME/.rlvm/$SAVEDIR"
+    ln -s "$GAMEDIR/saves" "$HOME/.rlvm/$SAVEDIR"
+done
+
+export LD_LIBRARY_PATH="$rlvm_dir/libs":$LD_LIBRARY_PATH
+if [ "$LIBGL_FB" != "" ]; then
+  export SDL_VIDEO_GL_DRIVER="$rlvm_dir/gl4es/libGL.so.1"
```
Comment (Cebion) [inline ports/kanon/Kanon.sh:63]:
```suggestion
  export SDL_VIDEO_GL_DRIVER="$rlvm_dir/gl4es/libGL.so.1"
  export LD_LIBRARY_PATH="$rlvm_dir/gl4es:$LD_LIBRARY_PATH"
```

File: ports/littlebusters/Little Busters!.sh [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+$ESUDO mount "$rlvm_file" "$rlvm_dir"
+PATH="$rlvm_dir:$PATH"
+
+# Create config dir
+rm -rf "$HOME/.rlvm/KEY_リトルバスターズ！"
+ln -s "$GAMEDIR/saves" "$HOME/.rlvm/KEY_リトルバスターズ！"
+
+export LD_LIBRARY_PATH="$rlvm_dir/libs":$LD_LIBRARY_PATH
+if [ "$LIBGL_FB" != "" ]; then
+  export SDL_VIDEO_GL_DRIVER="$rlvm_dir/gl4es/libGL.so.1"
```
Comment (Cebion) [inline ports/littlebusters/Little Busters!.sh:60]:
```suggestion
  export SDL_VIDEO_GL_DRIVER="$rlvm_dir/gl4es/libGL.so.1"
  export LD_LIBRARY_PATH="$rlvm_dir/gl4es:$LD_LIBRARY_PATH"
```

File: ports/planetarian/Planetarian.sh [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+$ESUDO mount "$rlvm_file" "$rlvm_dir"
+PATH="$rlvm_dir:$PATH"
+
+# Create config dir
+rm -rf "$HOME/.rlvm/KEY_planetarian_ME"
+ln -s "$GAMEDIR/saves" "$HOME/.rlvm/KEY_planetarian_ME"
+
+export LD_LIBRARY_PATH="$rlvm_dir/libs":$LD_LIBRARY_PATH
+if [ "$LIBGL_FB" != "" ]; then
+  export SDL_VIDEO_GL_DRIVER="$rlvm_dir/gl4es/libGL.so.1"
```
Comment (Cebion) [inline ports/planetarian/Planetarian.sh:60]:
```suggestion
  export SDL_VIDEO_GL_DRIVER="$rlvm_dir/gl4es/libGL.so.1"
  export LD_LIBRARY_PATH="$rlvm_dir/gl4es:$LD_LIBRARY_PATH"
```

File: ports/tomoyo_after/Clannad Tomoyo After.sh [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+$ESUDO mount "$rlvm_file" "$rlvm_dir"
+PATH="$rlvm_dir:$PATH"
+
+# Create config dir
+rm -rf "$HOME/.rlvm/KEY_智代アフター_EN_ALL"
+ln -s "$GAMEDIR/saves" "$HOME/.rlvm/KEY_智代アフター_EN_ALL"
+
+export LD_LIBRARY_PATH="$rlvm_dir/libs":$LD_LIBRARY_PATH
+if [ "$LIBGL_FB" != "" ]; then
+  export SDL_VIDEO_GL_DRIVER="$rlvm_dir/gl4es/libGL.so.1"
```
Comment (Cebion) [inline ports/tomoyo_after/Clannad Tomoyo After.sh:60]:
```suggestion
  export SDL_VIDEO_GL_DRIVER="$rlvm_dir/gl4es/libGL.so.1"
  export LD_LIBRARY_PATH="$rlvm_dir/gl4es:$LD_LIBRARY_PATH"
```

**Review by Cebion — CHANGES_REQUESTED** [review]

**Review by Cebion — APPROVED** [review]

**Review by kloptops — APPROVED** [review]

**Pull request merged**